File: src/ChainMetrics.UI/app/types/product.py
from pydantic import BaseModel, Field


class Product(BaseModel):
    id: str = Field(..., alias="id")
    category: str = Field(..., alias="category")
    weight: float = Field(..., alias="weight")
    length: float = Field(..., alias="length")
    height: float = Field(..., alias="height")
    width: float = Field(..., alias="width")


class PerCategoryDTO(BaseModel):
    category: str = Field(..., alias="category")
    quantity_products: int = Field(..., alias="quantityProducts")
    total_order_price: float = Field(..., alias="totalOrderPrice")

File: src/ChainMetrics.Domain/Entities/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ChainMetrics.Domain.Entities;

[Table("df_Products")]
public class Product
{
    [Key]
    [Column("product_id")]
    public required string Id { get; set; }

    [Column("product_category_name")]
    public required string Category { get; set; }

    [Column("product_weight_g")]
    public double Weight { get; set; }

    [Column("product_length_cm")]
    public double Length { get; set; }

    [Column("product_height_cm")]
    public double Height { get; set; }

    [Column("product_width_cm")]
    public double Width { get; set; }
}
